documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,1182 @@
+# Summary
+A new API for generic maths in Silk.NET.Maths 3.X, leveraging modern .NET features such as `INumber<T>` and vectorization.
+
+This API aims to replace the existing implementation of Silk.NET.Maths.
+
+# Contributors
+- Maxine H (@uwx)
+- Andrew Davis (@Curin)
+
+# Current Status
+- [x] Proposed
+- [ ] Discussed with API Review Board (ARB)
+- [ ] Approved
+- [ ] Implemented
+
+# Design Decisions
+- This proposal should replace the 2.X implementation of `Silk.Net.Maths`, matching `System.Numerics` where possible, with concessions for design oversights in that api.
+- This proposal assumes no knowledge of the 2.x Math library.
+- Text herein marked **INFORMATIVE** does not form a normative part of this proposal, and is for background only.
+- Within this proposal, the key words **must**, **required**, **shall**, **should**, **recommended**, **may**, **could**, and **optional** are to be interpreted as described in [RFC 2119 - Key words for use in RFCs to Indicate Requirement Levels](https://www.ietf.org/rfc/rfc2119.txt). The additional key word **optionally** is an alternate form of **optional**, for use where grammatically appropriate. These key words are highlighted in the proposal for clarity.
```

Comment:
I don't think this is applied correctly, if at all?
I'm a fan of this if you are motivated to put in the effort, but as we generally don't expect proposals to go unchanged through more than a handful of contributors I'm fine not doing this also.
Marking it and then not doing it isn't great imo.

Reply:
Fixed up some of the language to better match this. This was mostly included for the Informative and optional text, but I do realize some of my other phrasing didn't match, so I pulled it in line.